mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java

```diff
@@ -176,6 +185,51 @@ public RecordDaoImpl(final PostgresClientFactory postgresClientFactory) {
     )).map(queryResult -> toRecordCollectionWithLimitCheck(queryResult, limit));
   }
 
+  public Future<RecordCollection> getMatchedRecords(MatchField matchedField, TypeConnection typeConnection, int offset, int limit, String tenantId) {
```

Comment:
This method is written similar to the existing method _getRecords()_ that was used earlier. I just added an innerJoin to the partition of _marc_indexers_ table